/*
 * Copyright (c) 2003, 2011, Oracle and/or its affiliates. All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice (including the next
 * paragraph) shall be included in all copies or substantial portions of the
 * Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 * DEALINGS IN THE SOFTWARE.
 */

Allow xset dpms settings to override .xscreensaver.
---
 driver/timers.c       |    3 +++
 driver/xscreensaver.c |    4 ++++
 2 files changed, 7 insertions(+), 0 deletions(-)

diff --git driver/timers.c driver/timers.c
--- driver/timers.c
+++ driver/timers.c
@@ -1549,6 +1549,8 @@ watchdog_timer (XtPointer closure, XtIntervalId *id)
 
   /* If the DPMS settings on the server have changed, change them back to
      what ~/.xscreensaver says they should be. */
+  /* mali- No!! if someone uses xset then let the changes be picked by
+   * xscreensaver...do not have to regulate everything from .xscreensaver.
   sync_server_dpms_settings (si->dpy,
                              (p->dpms_enabled_p  &&
                               p->mode != DONT_BLANK),
@@ -1556,6 +1558,7 @@ watchdog_timer (XtPointer closure, XtIntervalId *id)
                              p->dpms_suspend / 1000,
                              p->dpms_off / 1000,
                              False);
+  **/
 
   if (si->screen_blanked_p)
     {
diff --git driver/xscreensaver.c driver/xscreensaver.c
--- driver/xscreensaver.c
+++ driver/xscreensaver.c
@@ -1587,6 +1587,10 @@ main (int argc, char **argv)
   init_sigchld ();
 
   disable_builtin_screensaver (si, True);
+/*****
+  mali- this may need to be taken out in future if it hinders user
+  setting dpms values through xset.
+ *****/
   sync_server_dpms_settings (si->dpy,
                              (p->dpms_enabled_p  &&
                               p->mode != DONT_BLANK),

